### ISSUE TITLE
[MRESOLVER-236] Make it possible to resolve .asc on a 'fail' <checksumPolicy/> respository.

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resolver#1404]
**[Rod Widdowson](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rodwiddowson)** opened **[MRESOLVER-236](https://issues.apache.org/jira/browse/MRESOLVER-236?redirect=false)** and commented

(I'm guessing the resolver version - maven version is 3.8.4).

We accidently made one of our repositories `<checksumPolicy>fail</checksumPolicy>` some time ago and over the weekend an plugin we run started failing.

After some digging I discovered that the problem was when the code was programmatically trying to resolve a `jar.asc` file.  Eventually the code ended up in

```
org.eclipse.aether.internal.impl.Maven2RepositoryLayoutFactory line 196

        public List<Checksum> getChecksums {
            if ( isSignature( artifact.getExtension() ) )
            {
                return Collections.emptyList();
            }
```

This means that when the resolution hit the correct repository it (silently) failed the checksum check and moved on to the next one, eventually falling off the end of the list and failing to resolve.

Our work around is to set the `<checksumPolicy>` to warn (which is what it used to be).

'It would be nice if'
* The failure was slightly less quiet
* If it was possible - programmatically or by configuration - to resolve signatures from checksuming repositories.

I have not dived very deeply into the code - just enough to diagnose why our CI was exploding so spectacularly so I may have missed some trick in which case I apologise for asking for existing function

---

**Affects:** 1.7.3

**Issue Links:**
- [MRESOLVER-241](https://issues.apache.org/jira/browse/MRESOLVER-241) Resolver checksum calculation should be driven by layout
 (_**"is caused by"**_)
- [MRESOLVER-242](https://issues.apache.org/jira/browse/MRESOLVER-242) When no remote checksums provided by layout, transfer inevitably fails/warns
 (_**"is caused by"**_)

**Remote Links:**
- [GitHub Pull Request #159
](https://github.com/apache/maven-resolver/pull/159)

0 votes, 6 watchers

**Comment by jira-importer** [comment]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** commented

Fixed as Commited as 16b12f14eec8b6309a118c225f50f35292eb0a4f

I wont close out this one yet, I'd like to add an "IT" (demo snippet that runs as part of build, kinda integration test) for this one.

**Comment by jira-importer** [comment]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** commented

merged demo as well as https://github.com/apache/maven-resolver/commit/a3fa4010d2bc62395215d7319a54e4b3cfde0522

**Comment by jira-importer** [comment]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** commented

The fix will be present in maven-resolver 1.8.0 (with many other nice features). Resolver since 1.7.x is Java8. AFAIR, we planned to up resolver in Maven 3.9.x, maybe even to 1.8.0 if resolver released in time, unsure.

**Comment by jira-importer** [comment]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** commented

they are isolated as every job is using his own local repository
see https://github.com/apache/maven-jenkins-lib/blob/cb83da3826d940793902fd9964dd023133b4a4f5/vars/asfMavenTlpStdBuild.groovy#L112
with mavenLocalRepo:'.repository' the repository is related to the folder where the sources have been checkout.
sadly the stack trace doesn't give us much more details and especially about the file(s) causing problem
grhh why surefire has a default trimStackTrace true.
I will change Jenkins to run CI jobs with trimStackTrace false to get more trace
